src/drivers/rc_eredis.erl

```diff
@@ -0,0 +1,66 @@
+-module (rc_eredis).
+
+-behaviour (gen_server).
+
+-export ([start_link/1]).
+-export ([init/1, handle_cast/2, handle_call/3, handle_info/2, terminate/2]).
+
+start_link(#{name := Name} = Args) ->
+  gen_server:start_link({local, Name}, ?MODULE, Args, []);
+
+start_link(Args) ->
+  gen_server:start_link({local, eredis}, ?MODULE, Args, []).
+
+init(Args) ->
+  self() ! connect,
+  process_flag(trap_exit, true),
+  ValidArgs = #{host         => maps:get(host, Args, "127.0.0.1"),
+                port         => maps:get(port, Args, 6379),
+                password     => maps:get(password, Args, ""),
+                database     => maps:get(database, Args, 0),
+                reconnection => maps:get(reconnection, Args, {uniform, 500})},
+  {ok, maps:merge(#{retries => 0, state => no_connection}, ValidArgs)}.
+
+handle_info(connect, #{reconnection := {uniform, Time}} = State) ->
+  #{host     := Host,
+    port     := Port,
+    password := Pass,
+    database := Database} = State,
+  try
+    {ok, Pid} = eredis:start_link(Host, Port, Database, Pass, Time),
+    {noreply, State#{connection_ref => Pid, state => connected}}
+  catch
+    error:{badmatch, {error,{connection_error,Reason}}} ->
+      lager:info("Redis connection error:~p~n", [Reason]),
+      {noreply, State#{state => no_connection}}
+  end;
+
+handle_info(connect, #{reconnection := Type} = State) ->
+  lager:error("Wrong reconnect_type: ~p~n", [Type]),
+  {stop, bad_reconnect_type, State#{state => no_connection}};
+
+handle_info({'EXIT', _From, Reason}, #{reconnection := {uniform, Time}} = State) ->
+  #{retries := Retries} = State,
+  lager:info("Error trying to connect with redis~nRetries: ~p~nError:~p~n", [Retries, Reason]),
+  erlang:send_after(Time, self(), connect),
+  {noreply, State#{retries => Retries + 1, state => no_connection}};
+
+handle_info(_Msg, State) ->
+  {noreply, State}.
+
+handle_cast(_Msg, State) ->
+  {noreply, State}.
+
+handle_call(get_connection, _From, #{connection_ref := Pid} = State) ->
+  case sys:get_state(Pid) of
```

Comment:
since this a bit of black magic, it could use some comments explaining what's going on and why it was necessary